File: bot/src/main/java/by/yayauheny/tutochkatgbot/dto/backend/BuildingResponseDto.java
package by.yayauheny.tutochkatgbot.dto.backend;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;
import java.util.Map;
import java.util.UUID;

/**
 * Building response DTO
 */
public record BuildingResponseDto(
    @JsonProperty("id") UUID id,
    @JsonProperty("cityId") UUID cityId,
    @JsonProperty("name") String name,
    @JsonProperty("address") String address,
    @JsonProperty("buildingType") PlaceType buildingType,
    @JsonProperty("workTime") Map<String, Object> workTime,
    @JsonProperty("coordinates") LatLon coordinates,
    @JsonProperty("externalIds") Map<String, Object> externalIds,
    @JsonProperty("isDeleted") Boolean isDeleted,
    @JsonProperty("createdAt") Instant createdAt,
    @JsonProperty("updatedAt") Instant updatedAt
) {
    public String displayName() {
        if (name != null && !name.isBlank()) {
            return name;
        }
        if (address != null && !address.isBlank()) {
            return address;
        }
        return null;
    }
}

File: backend/src/main/kotlin/yayauheny/by/model/building/BuildingResponseDto.kt
package yayauheny.by.model.building

import io.swagger.v3.oas.annotations.media.Schema
import java.time.Instant
import java.util.UUID
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject
import by.yayauheny.shared.dto.LatLon
import by.yayauheny.shared.enums.PlaceType

@Serializable
@Schema(description = "Building response DTO")
data class BuildingResponseDto(
    @Contextual val id: UUID,
    @Contextual val cityId: UUID,
    val name: String?,
    val address: String,
    val buildingType: PlaceType?,
    val workTime: JsonObject?,
    val coordinates: LatLon,
    val externalIds: JsonObject?,
    val isDeleted: Boolean,
    @Contextual val createdAt: Instant,
    @Contextual val updatedAt: Instant
)
